# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/FileManagerBase.h
# Commit: ad7b08acadd6
# Full Hash: ad7b08acadd691fc86b52ba7f8babfaa57972c56
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-03-19 17:04:52
# Regressor Bug: 1623278
# File Overlap Count: 1
# Description:
#   Bug 1623278 - Use SafeRefPtr for FileInfo. r=dom-workers-and-storage-reviewers,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D65550
# ==============================================================================

diff -r bb094a6084b3 -r ad7b08acadd6 dom/indexedDB/FileManagerBase.h
--- a/dom/indexedDB/FileManagerBase.h	Thu Mar 19 09:32:53 2020 +0000
+++ b/dom/indexedDB/FileManagerBase.h	Thu Mar 19 09:33:31 2020 +0000
@@ -27,13 +27,13 @@
   using MutexType = StaticMutex;
   using AutoLock = mozilla::detail::BaseAutoLock<MutexType&>;
 
-  MOZ_MUST_USE RefPtr<FileInfo> GetFileInfo(int64_t aId) const {
+  MOZ_MUST_USE SafeRefPtr<FileInfo> GetFileInfo(int64_t aId) const {
     if (!AssertValid()) {
       // In release, the assertions are disabled.
       return nullptr;
     }
 
-    // TODO: We cannot simply change this to RefPtr<FileInfo>, because
+    // TODO: We cannot simply change this to SafeRefPtr<FileInfo>, because
     // FileInfo::AddRef also acquires the FileManager::Mutex.
     // This looks quirky at least.
     FileInfo* fileInfo;
@@ -42,16 +42,16 @@
       fileInfo = mFileInfos.Get(aId);
     }
 
-    return fileInfo;
+    return {fileInfo, AcquireStrongRefFromRawPtr{}};
   }
 
-  MOZ_MUST_USE RefPtr<FileInfo> CreateFileInfo() {
+  MOZ_MUST_USE SafeRefPtr<FileInfo> CreateFileInfo() {
     if (!AssertValid()) {
       // In release, the assertions are disabled.
       return nullptr;
     }
 
-    // TODO: We cannot simply change this to RefPtr<FileInfo>, because
+    // TODO: We cannot simply change this to SafeRefPtr<FileInfo>, because
     // FileInfo::AddRef also acquires the FileManager::Mutex.
     // This looks quirky at least.
     FileInfo* fileInfo;
@@ -68,7 +68,7 @@
       mFileInfos.Put(id, fileInfo);
     }
 
-    return fileInfo;
+    return {fileInfo, AcquireStrongRefFromRawPtr{}};
   }
 
   void RemoveFileInfo(const int64_t aId, const AutoLock& aFileMutexLock) {